File: backend/src/main/java/com/hsf/service/CustomerService.java
package com.hsf.service;

import com.hsf.entity.Customer;
import java.util.List;

public interface CustomerService {

    List<Customer> getAllCustomers();
    Customer getCustomerById(Long id);
    Customer createCustomer(Customer customer);
    Customer updateCustomer(Long id, Customer customer);
    void deleteCustomer(Long id, Customer customer);
}

File: frontend/src/services/customerService.ts
import api from './api';

export interface Customer {
  id?: number;
  number: number;
  name: string;
  language: string;
  notificationNumber: string;
  invoiceName: string;
  weekDays: string;
}

export const getAllCustomers = () => {
  return api.get<Customer[]>('/customers');
};

export const createCustomer = (customer: Customer) => {
  return api.post('/customers/add', customer);
};

export const updateCustomer = (id: number, customer: Customer) => {
  return api.put(`/customers/${id}`, customer)
}

export const deleteCustomer = (id: number, customer: Customer) => {
  return api.put(`/customers/${id}/delete`, customer)
}
